lib/ddr/derivatives/generator.rb

```diff
@@ -0,0 +1,19 @@
+module Ddr
+  module Derivatives
+    class Generator
```

Comment:
Since I assume this is an abstract class, we should document the API by adding the methods (raise NotImplementedError) and a comment above the class `# @abstract`.